src/services/global-config.ts

```diff
@@ -0,0 +1,39 @@
+import { Storage } from '@capacitor/storage'
+import { GlobalConfig } from '../global'
+import * as Sentry from '@sentry/browser'
+const globalConfigKey = 'globalConfig'
+
+export const globalConfigFetcher = async (companyCode: string): Promise<GlobalConfig | undefined> => {
+  const { value } = await Storage.get({ key: globalConfigKey })
+
+  let parsedConfig
+
+  if (value) {
+    try {
+      parsedConfig = JSON.parse(value)
+
+      const config = parsedConfig[companyCode] as GlobalConfig
+
+      if (config) return config
+    } catch (e) {
+      Sentry.captureException(e)
+    }
+  }
+
+  try {
+    const response = await fetch(`https://${companyCode}.workgrid.com/config/config.json`)
+
+    const globalConfig = (await response.json()) as GlobalConfig
+
+    const globalConfigToSave = parsedConfig ?? {}
+
+    globalConfigToSave[companyCode] = globalConfig
+
+    await Storage.set({ key: globalConfigKey, value: JSON.stringify(globalConfigToSave) })
+
+    return globalConfig
+  } catch (e) {
+    throw e
+    // No need to capture a Sentry error here because it may be an invalid company code and we don't need to see all these errors
+  }
```

Comment:
Two thoughts:
1) Fetch doesn't throw for a bad request - you have to check `response.ok` or the status code -- super annoying compared to other libs
2) Should we use two separate try/catch blocks, one for if there is a network problem (like a bad company code) and a second for if there is a problem stringifying/saving in storage & report that to sentry?

```suggestion
  try {
    const response = await fetch(`https://${companyCode}.workgrid.com/config/config.json`)
    if (!response.ok) throw new Error('Network request failed')

    const globalConfig = (await response.json()) as GlobalConfig

    const globalConfigToSave = parsedConfig ?? {}

    globalConfigToSave[companyCode] = globalConfig

    try {
      await Storage.set({ key: globalConfigKey, value: JSON.stringify(globalConfigToSave) })
    } catch (error) {
      Sentry.captureException(e)
      throw e
    }

    return globalConfig
  } catch (e) {
    throw e
    // No need to capture a Sentry error here because it may be an invalid company code and we don't need to see all these errors
  }
```

Reply:
Thanks for the suggestions. I'll get these changes integrated on the next push.